### ISSUE TITLE
Issues after latest cache-related optimization

### ISSUE THREAD

**Issue by unsystemizer** [CounterpartyXCP/counterparty-lib#846]
I can't say I'm sure this is related to the latest commits to `develop`, but I can say I haven't seen this error before, so it may or may not be related. I pressed CTRL+C when CP caught up (started initializing mempool cache), but not before it declared it's ready.

```
[2016-03-15 05:19:06][DEBUG] getrawtransaction_batch: txhash_list size: 8251 / raw_transactions_cache size: 3938 / # getrawtransaction calls: 8125
^C[2016-03-15 05:22:51][INFO] Received SIGINT.
[2016-03-15 05:22:51][INFO] Shutting down.
^C[2016-03-15 05:39:21][INFO] Received SIGINT.
[2016-03-15 05:39:21][INFO] Shutting down.
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 38, in _python_exit
    t.join()
  File "/usr/lib/python3.4/threading.py", line 1051, in join
    self._wait_for_tstate_lock()
  File "/usr/lib/python3.4/threading.py", line 1067, in _wait_for_tstate_lock
    elif lock.acquire(block, timeout):
  File "/usr/local/lib/python3.4/dist-packages/counterparty_lib-9.54.0-py3.4.egg/counterpartylib/server.py", line 43, in sigterm_handler
    sys.exit(0)
SystemExit: 0
$ date
Tue Mar 15 05:39:46 GMT 2016
```

I restarted and let it run this time and noticed it just keeps consuming virtual memory. Although Bitcoin Core is keeping up (402733 at the moment), CP hasn't moved since it caught up at 402724.
Virtual Memory used by counterparty-server increased to 1.39 GB quickly.

```
2016-03-15-T06:11:58+0000 [INFO] Block: 402722 (154.73s, hashes: L:da0df / TX:b7219 / M:f4196)
2016-03-15-T06:12:20+0000 [INFO] Block: 402723 (21.25s, hashes: L:37fb8 / TX:6119a / M:596ff)
2016-03-15-T06:14:24+0000 [INFO] Block: 402724 (124.19s, hashes: L:ef5ed / TX:dd742 / M:5ba9d)
$ date
Tue Mar 15 07:26:29 GMT 2016
$ top -b | grep counter
23099 xcp       20   0 1321196 914084    828 S   0.0 91.7  14:50.58 counterparty-se
23099 xcp       20   0 1321196 914084    828 S   0.0 91.7  14:50.58 counterparty-se
23099 xcp       20   0 1321196 914084    828 S   0.0 91.7  14:50.58 counterparty-se
...
23099 xcp       20   0 1357060 914644    488 S   0.0 91.8  14:55.07 counterparty-se
...

23099 xcp       20   0 1394888 914088      0 S   0.0 91.7  14:59.57 counterparty-se

```

To summarize: it works fine (fast) when it's catching up; once it's caught up, it stops keeping up (0 disk I/O for many minutes) and it seems it leaks memory. It also can't exit cleanly within 10 mins of CTRL+C.
##

**Comment by unsystemizer** [comment]
Not seeing this since v9.55
